### PR TITLE
[teammgr]: Validate port channel table keys

### PR DESCRIPTION
**Pull request by ashutosh-agrawal**
**What I did**

Validate port-channel and member names where `teammgrd` consumes LAG, LAG-member, and port-state entries. The checks use `swss::isInterfaceNameValid()` from sonic-swss-common.

Member keys are parsed in one helper and are accepted only when they contain exactly one port-channel name and one member name. The same validation is applied before interface lookup and LAG member add/remove operations. Names passed to `teamd`, `teamdctl`, and interface commands are quoted, and the `ifreq` copy is length-bounded.

Valid port-channel and member entries continue through the existing lifecycle path unchanged.

**Why I did it**

`teammgrd` receives compound keys from Redis and later uses their components in interface operations and command arguments. Validating both components at the consumer boundary prevents malformed keys from being partially parsed, while the shared helper keeps the accepted interface syntax consistent across SWSS daemons.

**How I verified it**

- Built `tests_teammgrd` in the Bookworm environment.
- All 14 tests passed.
- Added coverage for invalid LAG names, malformed member keys, and quoted member-removal commands.

**Details if related**

Depends on the shared validator added by [sonic-swss-common #1240](https://github.com/sonic-net/sonic-swss-common/pull/1240), which is merged. This PR contains only the `teammgrd` consumer changes.

### CODE REVIEW

**Comment by ashutosh-agrawal** [comment]
Temporarily paused until the shared interface-name helper is available on master.

File: cfgmgr/teammgr.cpp [hunk 1]
```diff
@@ -25,6 +27,28 @@
 using namespace std;
 using namespace swss;
 
+namespace
+{
+
+bool parseLagMemberKey(const string &key, string &lag, string &member)
+{
+    auto tokens = tokenize(key, config_db_key_delimiter);
```
Comment (qiluo-msft) [inline cfgmgr/teammgr.cpp:35]: LGTM. One non-blocking edge case: `tokenize()` drops the final empty component, so `PortChannel1|Ethernet0|` passes this check as the normal member pair. Check there's exactly one separator and add a trailing-delimiter test? Existing behavior rather than a new regression; doesn't block this hardening.

Comment (ashutosh-agrawal) [reply cfgmgr/teammgr.cpp:40]: Good catch. I now require exactly one member-key separator before tokenization and added a dedicated trailing-delimiter regression test in 3bc58c70.
